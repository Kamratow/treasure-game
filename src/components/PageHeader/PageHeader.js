import React from "react";
import PropTypes from "prop-types";

import classes from "./PageHeader.module.css";

const PageHeader = ({ headerText }) => {
  return (
    <>
      <h1 className={classes.PageHeader}>{headerText}</h1>
      <h2>Find three treasures!</h2>
    </>
  );
};

PageHeader.propTypes = {
  headerText: PropTypes.string
};

export default PageHeader;
