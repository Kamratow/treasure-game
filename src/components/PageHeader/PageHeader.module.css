.PageHeader {
  font-size: 4rem;
}
