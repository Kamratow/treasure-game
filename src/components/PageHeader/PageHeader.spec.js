import React from "react";
import { shallow } from "enzyme";

import PageHeader from "./PageHeader";

const defaultProps = {
  headerText: "Test Header"
};

describe("PageHeader", () => {
  it("renders correctly with default props", () => {
    const wrapper = shallow(<PageHeader {...defaultProps} />).debug();
    expect(wrapper).toMatchSnapshot();
  });

  it("correctly passes text to main header", () => {
    const newHeaderText = "New Test Header";
    const newProps = { ...defaultProps, headerText: newHeaderText };
    const wrapper = shallow(<PageHeader {...newProps} />);
    const mainHeader = wrapper.find("h1.PageHeader");
    expect(mainHeader.text()).toBe(newHeaderText);
  });
});
