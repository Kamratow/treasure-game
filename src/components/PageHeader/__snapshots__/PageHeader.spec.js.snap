// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader renders correctly with default props 1`] = `
"<Fragment>
  <h1 className=\\"PageHeader\\">
    Test Header
  </h1>
  <h2>
    Find three treasures!
  </h2>
</Fragment>"
`;
